dotnet-desktop-guide/net/wpf/documents/snippets/how-to-print-xps-files/csharp/MainWindow.xaml.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Printing;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace CodeSampleCsharp
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml.
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+
+            // Events.
+            btnPrint.Click += BtnPrint_Click;
+        }
+
+        private async void BtnPrint_Click(object sender, RoutedEventArgs e)
+        {
+            // Check that the folder exists.
+            if (!Directory.Exists(txtFolderPath.Text))
+            {
+                MessageBox.Show("First select a valid folder path.");
+                return;
+            }
+
+            // Check whether any XPS files exist in the folder.
+            string[] xpsFilePaths = Directory.GetFiles(path: txtFolderPath.Text, searchPattern: "*.xps");
+            if (xpsFilePaths.Length == 0)
+            {
+                MessageBox.Show("No XPS files found.");
+                return;
+            }
+
+            // Determine whether XPS validation is needed.
+            bool fastCopy = cbxValidateXps.IsChecked == false;
+
+            // Batch add a collection of XPS files to the print queue.
+            // Run asynchronously to avoid blocking the UI thread.
+            bool isAllPrinted = await BatchAddToPrintQueueAsync(xpsFilePaths, fastCopy);
+
+            // Show result message.
+            MessageBox.Show($"{(isAllPrinted ? "Added" : "Failed to add")} all documents to the print queue.");
+        }
+
+        // <SampleCode1>
+        /// <summary>
+        /// Batch add a collection of XPS documents to the print queue using the PrintQueue.AddJob method.
+        /// </summary>
+        /// <param name="xpsFilePaths">A collection of XPS documents.</param>
+        /// <returns>Whether all documents were added to the print queue.</returns>
+        public static async Task<bool> BatchAddToPrintQueueAsync(IEnumerable<string> xpsFilePaths, bool fastCopy = false)
+        {
+            bool isAllPrinted = true;
+
+            // Queue some work to run on the ThreadPool.
+            // Wait for completion without blocking the calling thread.
+            await Task.Run(() =>
+            {
+                if (fastCopy)
+                {
+                    isAllPrinted = BatchPrint(xpsFilePaths, fastCopy);
+                }
```

Comment:
This can be collapsed to save space
```suggestion
                if (fastCopy)
                    isAllPrinted = BatchPrint(xpsFilePaths, fastCopy);
```

Reply:
Done.